android/app/src/main/kotlin/com/chicagoroboto/features/main/MainActivity.kt

```diff
@@ -69,30 +68,37 @@ class MainActivity : AppCompatActivity(), SessionNavigator, SpeakerNavigator, Na
 
     private fun showView(viewId: Int): Boolean {
         val view: View? = when (viewId) {
-            R.id.action_schedule -> SessionDateView(this)
+            R.id.action_schedule -> SessionDateView(this).setupTabLayout(tabs)
             R.id.action_speakers -> SpeakerListView(this)
             R.id.action_location -> LocationView(this)
             R.id.action_info -> InfoView(this)
             else -> null
         }
-        if (view != null) {
+        return view?.let {
             content.removeAllViews()
             content.addView(view)
-            return true
-        }
-        return false
+            val title = when (view) {
```

Comment:
I'd rather see an interface implemented here so that, assuming the view implements it, we can simply get the title from the view.  This is basically hard coding big switch cases that isn't very flexible (I'm not really a fan of the one above, either, but that's out of scope).

Reply:
I created a new interface, very originally titled, `MainView` that has a single properties to get the `titleResId` to be set as the toolbar title.

Let me know if you want me to rename it to something more consistent 👍